File: frontend/apps/android/app/src/main/java/app/pantopus/android/ui/screens/support_trains/start_train/components/TrainChip.kt
@file:Suppress("PackageNaming", "MagicNumber")

package app.pantopus.android.ui.screens.support_trains.start_train.components

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.platform.testTag
import androidx.compose.ui.unit.dp
import app.pantopus.android.ui.theme.PantopusColors
import app.pantopus.android.ui.theme.PantopusIcon
import app.pantopus.android.ui.theme.PantopusIconImage
import app.pantopus.android.ui.theme.PantopusTextStyle
import app.pantopus.android.ui.theme.Radii
import app.pantopus.android.ui.theme.Spacing

/**
 * A12.11 — Warm-amber "Support train" identity pill anchoring step 1 of
 * the wizard. Uses the warm-amber identity pillar so the chip reads as the
 * same accent the shell paints the progress rail and primary CTA in.
 */
@Composable
internal fun TrainChip() {
    Row(
        modifier =
            Modifier
                .clip(RoundedCornerShape(Radii.pill))
                .background(PantopusColors.warmAmberBg)
                .padding(horizontal = Spacing.s3, vertical = Spacing.s1)
                .testTag("startSupportTrainChip"),
        verticalAlignment = Alignment.CenterVertically,
        horizontalArrangement = Arrangement.spacedBy(Spacing.s1),
    ) {
        PantopusIconImage(
            icon = PantopusIcon.Heart,
            contentDescription = null,
            size = 11.dp,
            tint = PantopusColors.warmAmber,
        )
        Text(
            text = "SUPPORT TRAIN",
            style = PantopusTextStyle.overline,
            color = PantopusColors.warmAmber,
        )
    }
}

File: frontend/apps/ios/Pantopus/Features/SupportTrains/StartTrain/Components/TrainChip.swift
//
//  TrainChip.swift
//  Pantopus
//
//  A12.11 — Warm-amber "Support train" identity pill that anchors the top
//  of the wizard's first step. Uses the warm-amber identity pillar
//  (`warmAmber` / `warmAmberBg`) so the chip reads as the same accent the
//  shell paints the progress rail and primary CTA in.
//

import SwiftUI

/// Small uppercase identity chip — "SUPPORT TRAIN" in the warm-amber
/// pillar. Mirrors the design's `TrainChip`.
struct TrainChip: View {
    var body: some View {
        HStack(spacing: 5) {
            Icon(.heart, size: 11, strokeWidth: 2.4, color: Theme.Color.warmAmber)
            Text("SUPPORT TRAIN")
                .font(.system(size: 10, weight: .bold))
                .foregroundStyle(Theme.Color.warmAmber)
                .kerning(0.4)
        }
        .padding(.horizontal, 10)
        .padding(.vertical, Spacing.s1)
        .background(Theme.Color.warmAmberBg)
        .clipShape(Capsule())
        .accessibilityElement(children: .combine)
        .accessibilityLabel("Support train")
        .accessibilityIdentifier("startSupportTrainChip")
    }
}

#Preview {
    TrainChip()
        .padding(Spacing.s4)
        .background(Theme.Color.appBg)
}
